### ISSUE TITLE
Combine finance and approvals logic

### ISSUE THREAD

**Issue by joseph-palmer** [alan-turing-institute/rctab-api#33]
Approvals allow a certain amount of funds to be spent in a given period. We also have a finance table which separately captures the amount of funds that will be charged to a finance code. Approvals are accompanied by finance entries when an approval is for a project funding which has been approved. If no finance entry is given, it assumes that funding is covered from core. This can create problems when a finance entry is miss typed or forgotten.

I propose we require all approvals to have a finance entry alongside them that has the same amount and start & end dates as the approval. Core funded subscriptions will have a finance entry made with the code 'CORE'.  This will ensure we are explicit about where spending is recovered from.

**Comment by Iain-S** [comment]
Agreed. In fact, I think that approvals could be removed entirely. That would require:

1. [ ] Replace all references to the approvals table with references to the finance table.
2. [ ] Remove approvals and allocations commands from CLI.
3. [ ] Decide how to handle allocations. Do we keep the allocations table?
4. [ ] Decide how to migrate existing database. Presumably, the alembic upgrade could insert into the finance table where there are missing rows.
5. [ ] [optional] Make it so that finance rows can start and end on any date. Otherwise, we're limited to whole months.
6. [ ] [optional] Rename all finance-related objects so that they are known as approvals.
